### ISSUE TITLE
Probably my bad

### ISSUE THREAD

**Issue by dstephens1980** [molu8bits/squid-filebeat-kibana#4]
I followed the install instructions without encountering any errors yet the dashboard won't populate with data.  Watching "live stream" in the logs shows that data is coming in from squid server through filebeat but we are seeing the following "squid.access.error.message" "Provided Grok expressions do not match field value:".  Here is a sample message that we see that error with, but it happens for everything coming in (IP masked for security):

1561001136.756 0 172.XXX.XXX.64 TAG_NONE/400 4437 NONE error:invalid-request - HIER_NONE/- text/html

OS: Centos 7.7
ElasticSearch: 7.3.2-1
Kibana: 7.3.2-1
Filebeat: 7.3.2-1

Please let me know if I need to provide more data.
Thank you,

**Comment by molu8bits** [comment]
Seems that the grok has some problem with this part of log "... error:invalid-request ..." - namely it finds "error" word and following ":invalid-request". 
I haven't seen this in my logs and current grok pattern doesn't recognize that. Can you check if all your logs consists of "error:invalid-request" phrases  ? and provide me few lines more (with and without "error:invalid-request") ?

**Comment by dstephens1980** [comment]
[squid.access.invalid-request.log](https://github.com/molu8bits/squid-filebeat-kibana/files/3784602/squid.access.invalid-request.log)
[squid.access.typical.log](https://github.com/molu8bits/squid-filebeat-kibana/files/3784603/squid.access.typical.log)
@molu8bits Thank you for your quick response.  I have attached logs for our typical traffic as well as one with the error:invalid-request.  And I already checked with our dev team, the "error:invalid-request" is expected to occur from time to time.

**Comment by molu8bits** [comment]
Seems that URLs in the logs are not full. I mean here that we have e.g.

> 1563080217.302      0 172.XXX.XXX.101 TAG_NONE/400 3906 GET / - HIER_NONE/- text/html
instead of:
> 1563080217.302`      0 172.XXX.XXX.101 TAG_NONE/400 3906 GET http://www.example.com/ - HIER_NONE/- text/html

The reason for that could be that your logging configuration for squid.conf is changed from default
Example how to enforce default configuration and allow to record the rest of url:
> logformat squid      %ts.%03tu %6tr %>a %Ss/%03>Hs %<st %rm %ru %[un %Sh/%<a %mt
> access_log /var/log/squid/access.log squid

The second reason for such behaviour could be (I am not sure) that squid is configured as a reverse proxy and full url is not recorder. For that case the grok patter should be enriched by the third pattern (inside pipeline.json)

> "%{NUMBER:squid.access.timestamp}%{SPACE}%{NUMBER:squid.access.response_time} %{IP:squid.access.src_ip} %{DATA:squid.access.squid_request_status}/%{NUMBER:squid.access.http_status_code} %{NUMBER:squid.access.reply_size_include_header} %{DATA:squid.access.http_method} %{NOTSPACE:squid.access.request_url} %{NOTSPACE:squid.access.user} %{DATA:squid.access.hierarchy_code}/(?:-|%{IP:squid.access.dst_ip}) %{NOTSPACE:squid.access.content_type}"

**Comment by dstephens1980** [comment]
Awesome!  I really appreciate your help with this.  I will let you know how it goes.

**Comment by dstephens1980** [comment]
@molu8bits First off, Thank you! The first option fixed the issue.  Adjusting the squid logs to make sure they were compliant was the fix we needed.  Would you mind if I asked another question, or would you prefer I open a new issue?  I am not sure if this is related to your plugin or not, I know this one turned out not to be.

**Comment by molu8bits** [comment]
You are welcome. Please continue questions here

**Comment by dstephens1980** [comment]
Thank you again :)  So we got our initial test squid tied into elastic with filebeats and your module.  We decided to add in two more squid servers, so I replicated what we did on the first one.  1 of the 2 new servers connects just fine.  The one we are having issue with attempts to connect but doesn't ever complete.  I have tested with telnet and the elastic server responds fine from the failing server.  filebeat is setup correctly following the setup guide from here.  I am gathering logs but wanted to make sure this is something you think might be caused by the module or if you think I should take this to the Filebeats devs directly.
